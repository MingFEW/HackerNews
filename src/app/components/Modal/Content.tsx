import React from 'react'
import 'twin.macro'

interface Props {
  children: React.ReactNode
}

const Content: React.FC<Props> = ({ children }) => {
  return <div tw="p-4 overflow-auto bg-card [height: calc(100vh - 59px - 48px)]">{children}</div>
}

export default Content
