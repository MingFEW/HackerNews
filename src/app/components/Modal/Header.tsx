import React from 'react'
import { BsArrowLeft } from 'react-icons/bs'
import tw from 'twin.macro'

interface Props {
  title: string
  onClose: () => void
}

const Header: React.FC<Props> = props => {
  const { title, onClose } = props
  return (
    <div css={[tw`bg-gradient-to-r from-indigo-500 via-purple-500 to-pink-500`, tw`flex items-center gap-4 p-4`]}>
      <button type="button" tw="cursor-pointer" onClick={onClose}>
        <BsArrowLeft tw="text-white text-2xl" />
      </button>
      <div tw="text-lg font-bold text-white">{title}</div>
    </div>
  )
}

export default Header
