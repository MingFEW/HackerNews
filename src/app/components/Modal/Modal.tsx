import ReactModal, { Props as ReactModalProps } from 'react-modal'

// Components
import ModalHeader from './Header'
import ModalContent from './Content'

ReactModal.setAppElement('#root')

/**
 * styles
 */
const baseStyles = {
  content: {
    top: '50%',
    left: '50%',
    right: 'auto',
    bottom: 'auto',
    marginRight: '-50%',
    transform: 'translate(-50%, -50%)',
    padding: '0px',
    outline: 'none',
    border: 'none',
    background: 'transparent',
    maxWidth: '768px',
    width: '100%',
    borderRadius: 16,
    zIndex: 1000
  },
  overlay: {
    backgroundColor: 'rgba(var(--color-card), 0.6)',
    backdropFilter: 'blur(22px)'
  }
}

interface ModalProps extends ReactModalProps {
  children: React.ReactNode
}

interface CompoundedComponent {
  Header: typeof ModalHeader
  Content: typeof ModalContent
}

const Modal: React.FC<ModalProps> & CompoundedComponent = props => {
  const { children, style, ...rest } = props
  const modalStyles = { ...baseStyles, ...style }
  return (
    <ReactModal tw="relative" style={modalStyles} {...rest}>
      {children}
    </ReactModal>
  )
}

Modal.Header = ModalHeader
Modal.Content = ModalContent

export default Modal
