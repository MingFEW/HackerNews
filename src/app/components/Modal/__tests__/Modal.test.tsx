import { createRenderer } from 'react-test-renderer/shallow'
import Content from '../Content'

import Modal from '../Modal'

const shallowRenderer = createRenderer()

describe('<Modal />', () => {
  it('should render and match the snapshot', () => {
    shallowRenderer.render(
      <Modal isOpen>
        <Content>content</Content>
      </Modal>
    )
    const renderedOutput = shallowRenderer.getRenderOutput()
    expect(renderedOutput).toMatchSnapshot()
  })
})
