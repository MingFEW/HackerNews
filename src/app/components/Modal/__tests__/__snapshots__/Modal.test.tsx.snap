// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should render and match the snapshot 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  closeTimeoutMS={0}
  contentElement={[Function]}
  isOpen={true}
  overlayElement={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  preventScroll={false}
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
  style={
    {
      "content": {
        "background": "transparent",
        "border": "none",
        "borderRadius": 16,
        "bottom": "auto",
        "left": "50%",
        "marginRight": "-50%",
        "maxWidth": "768px",
        "outline": "none",
        "padding": "0px",
        "right": "auto",
        "top": "50%",
        "transform": "translate(-50%, -50%)",
        "width": "100%",
        "zIndex": 1000,
      },
      "overlay": {
        "backdropFilter": "blur(22px)",
        "backgroundColor": "rgba(var(--color-card), 0.6)",
        "zIndex": 999,
      },
    }
  }
  tw="relative"
>
  <Content>
    content
  </Content>
</Modal>
`;
